treeherder/intermittents_commenter/commenter.py

```diff
@@ -63,41 +63,43 @@ def generate_bug_changes(self, startday, endday, alt_startday, alt_endday):
-
             if self.weekly_mode:
                 priority = self.assign_priority(priority, counts)
                 if priority == 2:
-                    bug_info, whiteboard = self.check_bug_info(bug_info, bug_id)
-                    # if fetch_bug_details fails (called in check_bug_info), None is returned
-                    if bug_info is not None:
-                        change_priority, whiteboard = self.check_needswork_owner(change_priority, bug_info, whiteboard)
+                    bug_info = self.fetch_bug_details(TRIAGE_PARAMS, bug_id) if not bug_info else bug_info
+                    # if fetch_bug_details fails, None is returned
+
+                    if bug_info:
+                        change_priority, change_whiteboard = self.check_needswork_owner(change_priority, bug_info)
 
                 # change [stockwell needswork] to [stockwell unknown] when failures drop below 20 failures/week
                 if (counts['total'] < 20):
-                    bug_info, whiteboard = self.check_bug_info(bug_info, bug_id)
-                    # if fetch_bug_details fails (called in check_bug_info), None is returned
-                    if whiteboard is not None:
-                        whiteboard = self.check_needswork(whiteboard)
+                    bug_info = self.fetch_bug_details(TRIAGE_PARAMS, bug_id) if not bug_info else bug_info
+                    # if fetch_bug_details fails, None is returned
+
+                    if bug_info:
+                        change_whiteboard = self.check_needswork(change_whiteboard or bug_info['whiteboard'])
 
                 if bug_id in top_bugs:
                     rank = top_bugs.index(bug_id)+1
             else:
-                bug_info, whiteboard = self.check_bug_info(bug_info, bug_id)
-                # if fetch_bug_details fails (called in check_bug_info), None is returned
-                if bug_info is not None:
-                    change_priority, whiteboard = self.check_needswork_owner(change_priority, bug_info, whiteboard)
+                bug_info = self.fetch_bug_details(TRIAGE_PARAMS, bug_id) if not bug_info else bug_info
+                # if fetch_bug_details fails, None is returned
+                if bug_info:
+                    change_priority, change_whiteboard = self.check_needswork_owner(change_priority, bug_info)
+
+            # recommend disabling when more than 150 failures tracked over 21 days
+            if alt_bug_stats[bug_id]['total'] >= 150:
+                bug_info = self.fetch_bug_details(TRIAGE_PARAMS, bug_id) if not bug_info else bug_info
+                # if fetch_bug_details fails, None is returned
+
+                if bug_info and not self.check_whiteboard_status(bug_info['whiteboard']):
+                    priority = 3
+                    change_whiteboard = self.update_whiteboard(bug_info['whiteboard'], '[stockwell disable-recommended]')
```

Comment:
We're passing bug_info['whiteboard'] into self.update_whiteboard rather than change_whiteboard because recommended-disabled should have precedence over any previous whiteboard changes anyways (please correct me if I'm wrong).

Reply:
That is correct.